engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java

```diff
@@ -158,6 +164,29 @@ private Either<Failure, String> evalDateExp(final Expression date, final long sc
         .map(optionalDate -> optionalDate.map(ZonedDateTime::toString).orElse(null));
   }
 
+  private Either<Failure, String> evalFormId(final Expression formIdExp, final long scopeKey) {
+    if (formIdExp == null) {
+      return Either.right(null);
+    }
+    final String formId =
```

Comment:
Can you explain better what you mean ?

Reply:
Instead of using `getOrElse`, let's `flatMap` the result, so if the `formId` is not available we basically return failure instead of querying the database with empty `formId`. Does this help?